Source/OxyPlot/Series/DataRange.cs

```diff
@@ -0,0 +1,111 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="DataRange.cs" company="OxyPlot">
+//   Copyright (c) 2014 OxyPlot contributors
+// </copyright>
+// <summary>
+//   Represents an interval defined by two doubles
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace OxyPlot.Series
+{
+    using System;
+
+    /// <summary>
+    /// Represents an interval defined by two doubles.
+    /// </summary>
+    public struct DataRange : ICodeGenerating
+    {
+        /// <summary>
+        /// The undefined data range.
+        /// </summary>
+        public static readonly DataRange Undefined = new DataRange(double.NaN, double.NaN);
+
+        private double? minimum;
+        private double? maximum;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DataRange"/> struct.
+        /// </summary>
+        /// <param name="min">lower bound.</param>
+        /// <param name="max">upper bound.</param>
+        public DataRange(double min, double max)
+        {
+            if (max < min)
```

Comment:
I'd prefer that we explicitly reject NaN values here (so that it is impossible to accidently construct an undefined interval), but I'd defer to @objorke for that decision. The rest of the implementation looks fine wrt. to the current behaviour of accepting NaNs.

Reply:
Done.